src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringMethodCharOverloadWithSingleCharacters.cs

```diff
@@ -56,6 +56,10 @@ public abstract class UseStringMethodCharOverloadWithSingleCharacters : Diagnost
             nameof(string.IndexOf),
             nameof(string.LastIndexOf));
 
+        private INamedTypeSymbol? _stringComparisonType;
+        private ISymbol? _ordinalStringComparisonSymbol;
+        private ISymbol? _invariantCultureStringComparisonSymbol;
```

Comment:
Analyzers should not have state. When you resolve these in compilation start, you should then capture them and pass them to `AnalyzeOperation` instead of storing to fields.

Reply:
Done. (That's a lot of variables to pass, wondering if wrapping them in an ad-hoc class is better, similar to the dependencies class pattern)